### PR TITLE
Actions for examples + goodbye pipelines

### PR DESCRIPTION
**Pull request by mloubout**
WIP

Moved leftover pipeline to actions:

- mpi tuto int pytest-mpi
- examples into example action

Example action uses conda, test already use some of examples so no need for extra pip.

### CODE REVIEW

File: .github/workflows/pytest-core-mpi.yml [hunk 1]
```diff
@@ -45,6 +45,12 @@ jobs:
       run: |
         pytest --cov --cov-config=.coveragerc --cov-report=xml -m parallel tests/
 
+    - name: Test mpi notebook
+      run : |
+        ipcluster start --profile=mpi -n 4 --daemon
+        py.test --nbval examples/mpi
```
Comment (FabioLuporini) [inline .github/workflows/pytest-core-mpi.yml]: I suspect this is gonna fail. See #859

File: .github/workflows/examples.yml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+       environment-file: environment.yml
+       python-version: 3.7
+       auto-activate-base: false
+
+    - name: Install dependencies
+      # See setup-miniconda README, necessary to activate environment
+      shell: bash -l {0}
+      run: |
+        pip install -e .
+        pip install matplotlib
```
Comment (FabioLuporini) [inline .github/workflows/examples.yml]: actually necessary?

File: .github/workflows/examples.yml [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        python examples/seismic/tti/tti_example.py -a basic
+        python examples/seismic/tti/tti_example.py -a basic --noazimuth
+
+    - name: Seismic elastic examples
+      shell: bash -l {0}
+      run: |
+        py.test examples/seismic/elastic/elastic_example.py
+        py.test examples/seismic/viscoelastic/viscoelastic_example.py
+
+    - name: Linear algebra example
```
Comment (FabioLuporini) [inline .github/workflows/examples.yml]: nitpicking: I'd use "exampleS" (there are quite a few in linalg.py ; also, we may add more in the future)

Comment (FabioLuporini) [reply .github/workflows/examples.yml]: maybe more than "linear algebra"...how about "linear and non-linear operators" (in perspective of extending this? I'm thinking for example about adding convolutions and some more exotic examples/operators)

File: .github/workflows/examples.yml (hunk 2 above)
Comment (mloubout) [reply .github/workflows/examples.yml]: yes, matplotlib is in optionals but needed by the cfd test/example

File: .github/workflows/examples.yml (hunk 3 above)
Comment (mloubout) [reply .github/workflows/examples.yml]: >adding convolutions 

That's still linear algebra ;)
Only one example file so can add an s hen there is more or rename it but though was accurate for what's currently in it

**Comment by mloubout** [comment]
>they should probably stay in until #1085 is in ?

Yes that's the plan

File: .github/workflows/examples.yml [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+       auto-activate-base: false
+
+    - name: Install dependencies
+      # See setup-miniconda README, necessary to activate environment
+      shell: bash -l {0}
+      run: |
+        pip install -e .
+        pip install matplotlib
+
+    - name: Seismic acoustic examples
```
Comment (FabioLuporini) [inline .github/workflows/examples.yml]: Question, for another day...technically speaking, wouldn't it be more correct to call this "isotropic" or "iso-acoustic" ?

File: .github/workflows/examples.yml [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    - name: Seismic elastic examples
+      shell: bash -l {0}
+      run: |
+        py.test examples/seismic/elastic/elastic_example.py
+        py.test examples/seismic/viscoelastic/viscoelastic_example.py
+
+    - name: Linear algebra example
+      shell: bash -l {0}
+      run: |
+        python examples/misc/linalg.py mat-vec mat-mat-sum transpose-mat-vec
```
Comment (FabioLuporini) [inline .github/workflows/examples.yml]: why explicit list ?

File: .github/workflows/examples.yml (hunk 4 above)
Comment (mloubout) [reply .github/workflows/examples.yml]: isotropic-acoustic is more accurate

File: .github/workflows/examples.yml (hunk 5 above)
Comment (mloubout) [reply .github/workflows/examples.yml]: That's how it was in pipelines

**Comment by mloubout** [comment]
@fablup this is ready to go and doesn't really touch anything so can ust stay there open until #1085 is done I don't mind

**Comment by FabioLuporini** [comment]
@mloubout 

>  this is ready to go

agreed

> so can ust stay there open until #1085 is done I don't mind

yes, let's give it a few more days (@georgebisbas ), it shouldn't take much

File: environment.yml [hunk 6]
```diff
@@ -3,31 +3,6 @@ channels:
...
-  - sphinx
-  - sphinx_rtd_theme
-  - distributed>=1.27
-  - cython
-  - click
-  - codecov
-  - pytest-cov
-  - multidict
-  - frozendict
-  - nb_conda_kernels
```
Comment (FabioLuporini) [inline environment.yml]: this doesn't appear in requirements.txt ?

the following also are missing (not sure why):
- cython
- sphinx_rtd_theme ( I think necessary for rendering of documentation?)
- sphinx
- parso>=0.1.0 ??

Comment (mloubout) [reply environment.yml]: Gonna assume old unused dependency because everything works on all os/env and platforms

**Pull request merged**